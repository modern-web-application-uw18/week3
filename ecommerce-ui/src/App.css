.App {
  text-align: center;
  align-content: center;
  font-family: "Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Open Sans","Helvetica Neue",sans-serif;
  margin: auto;
  color: green;
}

.App-header-container {
  display: flex;
  flex-wrap: nowrap;
  height: 100px;
  width: 1050px;
  background-color: white;
  padding-top: 20px;
  margin: auto;
  margin-bottom: 20px;
}

.App-logo {
  height: 80px;
  margin-right: 50px;
}

.App-header-line {
  height: 25px;
  border-top: 1.5px;
  border-left: 0px;
  border-right: 0px;
  border-bottom: 0px;
  border-top-color: limegreen;
  border-style: solid;
  width: 1100px;
  margin: auto;
}

.App-title {
  font-size: 2em;
  color: limegreen;
  font-weight: bolder;
  margin-right: 300px;
}

.App-total {
  font-size: 1.4em;
  color: limegreen;
  font-weight: 500;
  height: 100px;
  width: 125px;
  margin-top: 30px;
}

.App-cart-container {
  display: flex;
  flex-wrap: nowrap;
  flex-direction: column;
  height: 100px;
  width: 60px;
  color: limegreen;
  font-weight: 700;
  background-color: white;
  margin-left: 20px;
  margin-bottom: 20px;
}

.App-ShoppingCart {
  margin-left: 20px;
}

.App-cart-title{
  height: 50px;
  width: 1050px;
  text-align: left;
  font-size: 20px;
  font-weight: 700;
  margin-left: 25px;
}

.App-intro {
  font-size: large;
  font-weight: bolder;
  font-size: 1.5em;
  margin: auto;
  margin-bottom: 50px;
}

.App-homes-list {
  display: flex;
  flex-wrap: wrap;
  margin: auto;
  height: 300px;
  width: 1050px;
  align-items: center;
  background-color: white;
}
