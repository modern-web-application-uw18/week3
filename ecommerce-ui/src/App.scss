body {
  font-family: sans-serif;
}

strong {
  font-weight: bold;
}

.btn {
  color: white;
  font-size: 1em;
  background-color: rgb(255, 90, 95);
  width: 100%;
  border-radius: 5px;
  border-color: rgb(255, 90, 95);
  line-height: 2em;
}

img {
  display: block;
  width: 100%;
}
