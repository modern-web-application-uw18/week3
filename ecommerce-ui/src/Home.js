import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import airbnbs from './airbnbs.json';
import BnbListings from './BnbListings/BnbListings';

class Home extends Component {

  render() {
    return (
      <section className="page-content">
        <h2 className="section-heading">Today&apos;s featured BNBs</h2>
        <div className="page-content--container">
          {airbnbs.map((airbnbs, idx) => {
            return <BnbListings key={idx} bnb={airbnbs} />;
          })}
        </div>
      </section>
    );
  }
}

export default Home;
