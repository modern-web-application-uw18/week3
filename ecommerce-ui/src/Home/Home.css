* {
    box-sizing: border-box;
}

.Home-flex {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    width: 330px;
    padding: 8px;
}

.Home-horizontal-flex {
    display: flex;
    flex-wrap: nowrap;
}

.Home-container {
    height: 275px;
    max-height: 275px;
    width: 345px;
    max-width: 345px;
    background-color: white;
    border: lightgray;
    border-width: 2px;
    border-style: solid;
    margin: 10px;
}

.Home-image-container {
    width: 330px;
}

.Home-image-container img {
    width: 330px;    padding-left: 20px;
    padding-right: 20px;
    overflow: hidden;
}

.Bolded {
    font-weight: bold;
}

.Home-Green-button {
    background-color: darkgreen;
    border: none;
    color: white;
    padding: 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
    font-weight: 700;
}

.Home-Gray-button {
    background-color: gray;
    border: none;
    color: black;
    padding: 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
    font-weight: 700;
}