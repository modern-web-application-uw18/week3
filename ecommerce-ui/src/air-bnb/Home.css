* {
    box-sizing: border-box;
}

.Home {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    width: 330px;
    margin: 8px;
}

.Home-image-container {
    width: 330px;
}

.Home-image-container img {
    width: 330px;
    overflow: hidden;
}

.Home-Green-button {
    background-color: darkgreen;
    border: none;
    color: white;
    padding: 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
    font-weight: 700;
}

.Home-Gray-button {
    background-color: gray;
    border: none;
    color: black;
    padding: 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    margin: 4px 2px;
    cursor: pointer;
    border-radius: 10px;
    font-weight: 700;
}
