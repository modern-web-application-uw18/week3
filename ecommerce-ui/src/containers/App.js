import React, { Component } from 'react';
import Layout from './Layout/Layout'
import Aux from '../hoc/Auxl';

class App extends Component {
  render() {
    return (
      <Aux>
        <Layout />
      </Aux>
    );
  }
}

export default App;
