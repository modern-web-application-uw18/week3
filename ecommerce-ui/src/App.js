import React, { Component } from 'react';

import './App.css';
import logo from './logo.svg';

import Location from './components/Location';
import airbandbs from './data/airbandbs.json';

// console.log({airbandbs});

class App extends Component {
  render() {

    return (

      <div className="App">
        <header className="App-header">
          <h4 className="App-cart"><a href="#">cart</a></h4>
          <img src={logo} className="App-logo" alt="logo" />
        </header>

        <section>
          {/* <Location airbandbs={airbandbs[0]} /> */}

          {airbandbs.map((airbandbs, idx) => {
            return <Location key={idx} airbandbs={airbandbs} />
          })}

        </section>
      </div>
    );
  }
}

export default App;
