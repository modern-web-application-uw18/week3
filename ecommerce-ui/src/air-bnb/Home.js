import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Description from './Description';
import Payment from './Payment';
import Host from './Host';
import Rating from './Rating';
import './Home.css';

class Home extends Component {

    render() {

        const {
            title,
            houseType,
            image,
            location,
            payment,
            host,
            rating
        } = this.props.home;

        return (
            <div className="Home" >
                <div className="Home-image-container" >
                    <img src={image} alt="" />
                </div>
                <Description title={title} houseType={houseType} location={location} />
                <Payment payment={payment} />
                <Host host={host} rating={rating} />
                <button onClick={this.props.addToCart}
                    className={this.props.isInCart(this) ? 'Grey-button' : 'Green-button'}>{this.props.isInCart(this) ? 'Added' : 'Add to Cart'}</button>
            </div >
        );
    }
}

Home.propTypes = {
    title: PropTypes.string.isRequired,
    houseType: PropTypes.string.isRequired,
    image: PropTypes.string,
    location: PropTypes.instanceOf(Location),
    payment: PropTypes.instanceOf(Payment),
    host: PropTypes.instanceOf(Host),
    rating: PropTypes.instanceOf(Rating),
};

Home.defaultProps = {
    title: 'No title :(',
    houseType: 'No house type :(',
};

export default Home;
