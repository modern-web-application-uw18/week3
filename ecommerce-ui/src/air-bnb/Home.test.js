import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import App from '../App';
import Home from './Home';
import testHome from '../json/test-home.json';

describe('Home', () => {

    function isInCart(home,e) {
        return false;
    }

    it('renders without crashing with data', () => {
        const div = document.createElement('div');
        const element = testHome.map((home, index) => {
            return <Home key={index} home={home} 
                    isInCart={e => isInCart(home,e)} />        
        });
        ReactDOM.render(element, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, description, image, etc.', () => {
        const element = testHome.map((home, index) => {
            return <Home key={index} home={home}
                    isInCart={e => isInCart(home,e)} />    
        });
        const component = TestRenderer.create(element);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
