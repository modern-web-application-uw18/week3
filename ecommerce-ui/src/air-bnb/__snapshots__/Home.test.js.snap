// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render with title, description, image, etc. 1`] = `
<div
  className="Home"
>
  <div
    className="Home-image-container"
  >
    <img
      alt=""
      src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
    />
  </div>
  <div
    className="Description"
  >
    <div>
      Entire House
       路 
      <span>
        Stockholm
        , 
        Sweden
      </span>
    </div>
    <div
      className="Home-Title"
    >
      Quaint 3-bedroom home in Gamla stan
    </div>
  </div>
  <div>
    $
    230
     per night 
    <span>
       路 
      Free cancellation
    </span>
  </div>
  <div
    className="Host"
  >
    Olivia
     
     路 
    <span>
      83
       reviews 路
      <span
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "direction": "ltr",
            "display": "inline-block",
          }
        }
      >
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <img
            alt=""
            src="gray-star.png"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt=""
              src="green-star.png"
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <img
            alt=""
            src="gray-star.png"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt=""
              src="green-star.png"
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <img
            alt=""
            src="gray-star.png"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt=""
              src="green-star.png"
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <img
            alt=""
            src="gray-star.png"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt=""
              src="green-star.png"
            />
          </span>
        </span>
        <span
          onClick={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "position": "relative",
            }
          }
        >
          <img
            alt=""
            src="gray-star.png"
          />
          <span
            style={
              Object {
                "display": "inline-block",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <img
              alt=""
              src="green-star.png"
            />
          </span>
        </span>
      </span>
    </span>
  </div>
  <button
    className="Home-Green-button"
    onClick={undefined}
  >
    Add to Cart
  </button>
</div>
`;
